src/components/htmlvideoplayer/plugin.js

```diff
@@ -583,28 +597,14 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
             if (currentAssRenderer) {
                 updateCurrentTrackOffset(offsetValue);
             } else {
-                var videoElement = self._mediaElement;
-                var mediaStreamTextTracks = getMediaStreamTextTracks(self._currentPlayOptions.mediaSource);
-
-                Array.from(videoElement.textTracks)
-                    .filter(function(trackElement) {
-                    // get showing .vtt textTacks
-                        return trackElement.mode === 'showing';
-                    })
-                    .forEach(function(trackElement) {
-
-                        var track = customTrackIndex === -1 ? null : mediaStreamTextTracks.filter(function (t) {
-                            return t.Index === customTrackIndex;
-                        })[0];
-
-                        if (track) {
-                            offsetValue = updateCurrentTrackOffset(offsetValue);
-                            setVttSubtitleOffset(trackElement, offsetValue);
-                        } else {
-                            console.log("No available track, cannot apply offset : " + offsetValue);
-                        }
-
-                    });
+                var trackElement = getTextTrack();
+                // if .vtt currently rendering
+                if (trackElement) {
```

Comment:
What happens here if "events" are returned?

Reply:
If ```getTextTrack();``` return **currentTrackEvents**, **currentTrackEvents**  is an array of events, which is handled by:
* ```if (Array.isArray(currentTrack))``` (line 631, function **setVttSubtitleOffset**).

Reply:
But isn't it truethy, thus going into 'then' clause while it shouldn't?

Reply:
hum... ok, I will try to rewrite this part in a more explicit way to avoid mistakes.